File: backend/src/main/java/com/shopstack/shopstack_backend/service/OrderService.java
package com.shopstack.shopstack_backend.service;

import com.shopstack.shopstack_backend.constant.OrderStatus;
import com.shopstack.shopstack_backend.dto.request.OrderRequest;
import com.shopstack.shopstack_backend.dto.response.OrderResponse;

import java.util.List;

public interface OrderService {

    OrderResponse createOrder(OrderRequest request);

    List<OrderResponse> getMyOrders();

    OrderResponse getOrderById(Long id);

    OrderResponse cancelOrder(Long id);

    OrderResponse requestReturn(Long id);

    // =========================
    // ORDER STATUS MANAGEMENT
    // =========================

    OrderResponse updateOrderStatus(
            Long id,
            OrderStatus newStatus
    );

    // =========================
    // WAREHOUSE ALLOCATION
    // =========================

    OrderResponse allocateWarehouse(
            Long orderId,
            Long warehouseId
    );
}

File: frontend/src/api/orderService.js
import api from "./api";

// =========================
// CREATE ORDER
// =========================

export const createOrder = async (orderData) => {

    const token =
        localStorage.getItem("token");

    return api.post(
        "/orders",
        orderData,
        {
            headers: {
                Authorization:
                    `Bearer ${token}`,
            },
        }
    );
};


// =========================
// GET MY ORDERS
// =========================

export const getMyOrders = async () => {

    const token =
        localStorage.getItem("token");

    return api.get(
        "/orders/my-orders",
        {
            headers: {
                Authorization:
                    `Bearer ${token}`,
            },
        }
    );
};


// =========================
// GET ORDER BY ID
// =========================

export const getOrderById = async (orderId) => {

    const token =
        localStorage.getItem("token");

    return api.get(
        `/orders/${orderId}`,
        {
            headers: {
                Authorization:
                    `Bearer ${token}`,
            },
        }
    );
};


// =========================
// CANCEL ORDER
// =========================

export const cancelOrder = async (orderId) => {

    const token =
        localStorage.getItem("token");

    return api.put(
        `/orders/${orderId}/cancel`,
        {},
        {
            headers: {
                Authorization:
                    `Bearer ${token}`,
            },
        }
    );
};


// =========================
// REQUEST RETURN
// =========================

export const requestReturn = async (orderId) => {

    const token =
        localStorage.getItem("token");

    return api.put(
        `/orders/${orderId}/return`,
        {},
        {
            headers: {
                Authorization:
                    `Bearer ${token}`,
            },
        }
    );
};
